src/dbnode/storage/types.go

```diff
@@ -339,6 +359,14 @@ type databaseNamespace interface {
 		start, end time.Time,
 	) ([][]xio.BlockReader, error)
 
+	// FetchIndexChecksum retrieves the index checksum for an ID batch for the
+	// block at time start.
+	FetchIndexChecksum(
+		ctx context.Context,
+		idBatch ident.ID,
```

Comment:
Isn't this a single ID? Should it just be `id ident.ID`? Also comment should change if so.